cmd/gardener-apiserver/app/gardener_apiserver.go

```diff
@@ -375,9 +382,63 @@ func (o *Options) Run(ctx context.Context) error {
 		return err
 	}
 
+	if err := server.GenericAPIServer.AddPostStartHook("bootstrap-public-info", func(_ genericapiserver.PostStartHookContext) error {
+		if err := createNamespaces(gardencorev1beta1.GardenerSystemPublicNamespace); err != nil {
+			return err
+		}
+
+		p := publicInfo{
+			Version:      version.Get().String(),
+			FeatureGates: fmt.Sprint(features.DefaultFeatureGate),
+		}
+
+		if len(o.ExtraOptions.WorkloadIdentityTokenIssuer) != 0 {
+			p.WorkloadIdentityIssuerURL = &o.ExtraOptions.WorkloadIdentityTokenIssuer
+		}
+
+		marshelledInfo, err := yaml.Marshal(p)
+		if err != nil {
+			return err
+		}
+
+		configMapName := "gardener-info"
+		gardenerAPIServerKey := "gardenerAPIServer"
```

Comment:
Nit: these can be constants and grouped
```suggestion
		const (
		    configMapName = "gardener-info"
		    gardenerAPIServerKey = "gardenerAPIServer"
		)
```